code/datums/ai/generic/generic_behaviors.dm

```diff
@@ -2,6 +2,7 @@
 /datum/ai_behavior/resist/perform(seconds_per_tick, datum/ai_controller/controller)
 	. = ..()
 	var/mob/living/living_pawn = controller.pawn
+	living_pawn.ai_controller.set_blackboard_key(BB_RESISTING, TRUE)
 	living_pawn.execute_resist()
 	finish_action(controller, TRUE)
```

Comment:
It seems like this _shouldn't_ complete until the do_after is finished anyway, although it seems from your testing that this is not the case?

Reply:
Yea in my testing they would cycle to other behaviors due to the resist chance not being 100%, so they'll want to wander around or try to pick up an item so they walk over and interrupt it.